jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsClientManager.kt

```diff
@@ -63,3 +71,22 @@ inline fun <reified T : SdkClient> Project.awsClient(): T {
         .getInstance()
         .getClient(accountSettingsManager.activeCredentialProvider, accountSettingsManager.activeRegion)
 }
+
+/**
+ * Used to override/add behavior during AWS SDK Client creation.
+ *
+ * Example usage to add a local development endpoint for a particular service:
+ *
+ * ```
+ * class MyDevEndpointCustomizer : AwsClientCustomizer {
+ *   override fun customize(builder: AwsClientBuilder<*, *>) {
+ *     if (builder is LambdaClientBuilder) {
+ *       builder.endpointOverride(URI.create("http://localhost:8888"))
+ *     }
+ *   }
+ * }
+ * ```
+ */
+interface AwsClientCustomizer {
+    fun customize(builder: AwsClientBuilder<*, *>)
```

Comment:
If we did want to enable localstack support, we probably need to pass in the connection settings (to get the region)

Reply:
k - good call - like it